services/src/main/java/org/keycloak/services/resources/account/AccountRestService.java

```diff
@@ -275,42 +257,53 @@ public Response getApplications() {
     @Produces(MediaType.APPLICATION_JSON)
     public Response getApplication(final @PathParam("clientId") String clientId) {
         checkAccountApiEnabled();
-        auth.require(AccountRoles.VIEW_APPLICATIONS);
+        auth.requireOneOf(AccountRoles.MANAGE_ACCOUNT, AccountRoles.VIEW_APPLICATIONS);
         ClientModel client = realm.getClientByClientId(clientId);
         if (client == null) {
             return Cors.add(request, Response.status(Response.Status.NOT_FOUND).entity("No client with clientId: " + clientId + " found.")).build();
         }
 
-        return Cors.add(request, Response.ok(modelToRepresentation(client))).build();
+        List<String> inUseClients = new LinkedList<>();
+        if(!session.sessions().getUserSessions(realm, client).isEmpty()) {
+            inUseClients.add(clientId);
+        }
+
+        List<String> offlineClients = new LinkedList<>();
+        if(session.sessions().getOfflineSessionsCount(realm, client) > 0) {
+            offlineClients.add(clientId);
+        }
+
+        return Cors.add(request, Response.ok(modelToRepresentation(client, inUseClients, offlineClients))).build();
```

Comment:
I think there's a potential security issue. If I got it right, this endpoint can return even clients that should be hidden from user like bearer-only clients or clients with no active session and no consents, etc.

Reply:
Updated to return not found for bearer-only clients and clients with no base url.

Reply:
@douglaspalmer Thanks!

Should we also hide Clients that are not in active session?
In other words, should we have some unified mechanism for filtering clients out that would ensure the results from `/applications` and `/applications/{client_id}` are consistent? `/applications/{client_id}` IMHO should return only the clients that are accessible through `/applications`. But maybe this is out of scope of this PR? WDYT @stianst @abstractj ?

Reply:
We should also have a test that would check this endpoint returns only the clients it's supposed to return.

Reply:
@vmuzikar I think this is out of scope. This endpoint has been updated out of necessity but isn't the focus of this work.

Reply:
I agree it's out of scope. We just need to decide what to do next.

Will this be fixed in a follow-up task, i.e. should we create a JIRA for it?

Or shall we remove this endpoint completely in this PR as it doesn't make sense anymore (as we discussed yesterday)? I believe this would be a better solution.

Reply:
This addressed in #6572, the end user can call /applications?name=clientId instead.

Reply:
@douglaspalmer Ok, thanks!